Change LandmarkAnnotationSet endpoint and add custom action

HarmvZ opened the pull request:
- Changes endpoint url from `/retina/api/...` to `/api/v1/retina/landmark-annotation/`
- Adds a custom action on the viewset for retrieving all annotations for a specific image provided in the querystring

Closes #1032

jmsmkn commented on app/grandchallenge/retina_api/views.py:
```diff
@@ -976,9 +978,38 @@ class ImageTextAnnotationViewSet(viewsets.ModelViewSet):
+            singlelandmarkannotation__image=image
+        )
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
```
Should this be done using a filter on the list view? https://www.django-rest-framework.org/api-guide/filtering/

HarmvZ replied:
Good point. I moved the functionality to the list view.

HarmvZ commented:
Refactored filtering:
 - Created filter backend for filtering annotations on owner or `retina_admin`
 - Filtered on query parameter in `get_queryset()` method
 - Added `prefetch_related` to improve performance since related `SingleLandmarkAnnotation` objects are used during serialization
